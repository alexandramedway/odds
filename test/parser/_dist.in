do <2, 3> | hello
do x = <5+6, 7+8> | (x) -> return x*x