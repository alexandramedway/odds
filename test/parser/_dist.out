[
    Do(
        Dist({
            min=Num_lit(2);
            max=Num_lit(3);
            dist_func=Id(hello)
        })
    );
    Do(
        Assign(
            x,
            Dist({
                min=Binop(Num_lit(5), Add, Num_lit(6));
                max=Binop(Num_lit(7), Add, Num_lit(8));
                dist_func=
                    Fdecl({
                        params=[x] ;
                        body=[] ;
                        return=Binop(Id(x), Mult, Id(x))
                    })
            })
        )
    )
]